diff -Naur garglk/garglk.ini garglk_new/garglk.ini
--- garglk/garglk.ini	2006-09-08 22:58:26.000000000 -0700
+++ garglk_new/garglk.ini	2007-08-27 19:59:11.000000000 -0700
@@ -22,7 +22,7 @@
 
 # ------------------------------------------------------------------
 
-scrollwidth	0	# set to 8 to make a nice scrollbar
+scrollwidth	8	# set to 8 to make a nice scrollbar
 scrollbg e0e0d0
 scrollfg c0c0b0
 
@@ -162,7 +162,7 @@
 # The name of a game file can also be used.
 #
 
-[ Agility Arun Magnetic Nitfol Scare ]
+[ Ifpe Legion Agility Arun Magnetic Nitfol Scare ]
 
 gcolor  0 ffffff 606060 # Normal
 gcolor  1 ffffff 606060 # Emphasized
