diff -Naur nitfol/nitfol-0.5/io.c nitfol_new/nitfol-0.5/io.c
--- nitfol/nitfol-0.5/io.c	Thu Oct 21 19:53:34 1999
+++ nitfol_new/nitfol-0.5/io.c	Sat Jul 26 09:25:46 2003
@@ -697,8 +697,11 @@
   end_line += start_line;
 
   o = glk_window_get_parent(window->win);
-  glk_window_set_arrangement(o, window->method,
-			     end_line, window->win);
+  glk_window_get_size(window->win, &winx, &winy);
+  if (!(window->method & winmethod_Above || window->method & winmethod_Below)
+      || winy != end_line)
+    glk_window_set_arrangement(o, window->method,
+			       end_line, window->win);
   glk_window_get_size(window->win, &winx, &winy);
 
   if(window->draw_callback) {
