diff -bur xglk/xg_win_textbuf.c xglk+hack/xg_win_textbuf.c
--- xglk/xg_win_textbuf.c       Thu Jun 27 00:10:36 2002
+++ xglk+hack/xg_win_textbuf.c  Thu Jun 27 02:27:50 2002
@@ -2392,7 +2392,7 @@
   char realch;

   /* ###### not perfect -- should be all typable chars */
-  if (ch < 32 || ch >= 127)
+  if (ch < 32 || (ch >= 127 && ch < 160) || ch > 255)
     ch = ' ';

   realch = ch;
diff -bur xglk/xg_win_textgrid.c xglk+hack/xg_win_textgrid.c
--- xglk/xg_win_textgrid.c      Sat Apr 15 20:56:45 2000
+++ xglk+hack/xg_win_textgrid.c Thu Jun 27 02:28:30 2002
@@ -1353,7 +1353,7 @@
   int ix, pos;

   /* ### not perfect -- should be all typable chars */
-  if (ch < 32 || ch >= 127)
+  if (ch < 32 || (ch >= 127 && ch < 160) || ch > 255)
     ch = ' ';

   realch = ch;
